crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -898,8 +931,15 @@ impl ReadOnlyDevice {
     /// replicated locally.
     pub async fn from_account(account: &ReadOnlyAccount) -> ReadOnlyDevice {
         let device_keys = account.device_keys().await;
-        ReadOnlyDevice::try_from(&device_keys)
-            .expect("Creating a device from our own account should always succeed")
+        let from_keys = ReadOnlyDevice::try_from(&device_keys)
+            .expect("Creating a device from our own account should always succeed");
+        ReadOnlyDevice { first_time_seen_ts: account.creation_local_time_ts().into(), ..from_keys }
```

Comment:
Let's not create 2 `ReadOnlyDevice` value. We can update the first one.

```suggestion
        let mut device = ReadOnlyDevice::try_from(&device_keys)
            .expect("Creating a device from our own account should always succeed");
        device.first_time_seen_ts = account.creation_local_time_ts().into();

        device
```